huaweicloud/services/ram/data_source_huaweicloud_ram_shared_resources.go

```diff
@@ -0,0 +1,183 @@
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/go-uuid"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/chnsz/golangsdk"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/common"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+const pageLimit = 20
+
+// @API RAM POST /v1/shared-resources/search
+func DataSourceRAMSharedResources() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: datasourceRAMSharedResourcesRead,
+		Schema: map[string]*schema.Schema{
+			"resource_owner": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"principal": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"resource_ids": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"resource_urns": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"resource_share_ids": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"resource_region": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"resource_type": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"shared_resources": {
+				Type:     schema.TypeList,
+				Elem:     sharedResourcesSchema(),
+				Computed: true,
+			},
+		},
+	}
+}
+
+func sharedResourcesSchema() *schema.Resource {
+	sc := schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"resource_urn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"resource_type": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"resource_share_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"status": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"created_at": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"updated_at": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+	return &sc
+}
+
+func datasourceRAMSharedResourcesRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	var mErr *multierror.Error
+
+	var (
+		getRAMSharedResourcesHttpUrl = "v1/shared-resources/search"
+		getRAMSharedResourcesProduct = "ram"
+	)
+	getRAMSharedResourcesClient, err := cfg.NewServiceClient(getRAMSharedResourcesProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating RAM client: %s", err)
+	}
+
+	getRAMSharedResourcesPath := getRAMSharedResourcesClient.Endpoint + getRAMSharedResourcesHttpUrl
+
+	getRAMSharedResourcesOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+	}
+	allSharedResources := make([]interface{}, 0)
+	nextMarker := "0"
```

Comment:
can we use "" to marker in the first API request.
```suggestion
	var nextMarker string
```

Reply:
No, the nextMarker field can not be empty.

Reply:
"0" is not used, please check it again.